### DISCUSSION TITLE
GUI Enhancements

### DISCUSSION THREAD

**Discussion by tangkong** [pcdshub/lightpath#131] [Ideas]
# Context
The gui has been plagued with minor and major problems, but there's room for more broad improvements

# Ideas / Approaches
- a global LCLS map/graph.  Could be simply lines that are bright or dark depending on if beam is getting through.  Could also click on those branches to expand the line and see the individual devices.  A BeamPath or similar object is a nice point of abstraction here, I think
- Various bugfixes
  - coloring of devices like slits

**Comment by ZLLentz** [comment]
The device type selectors on the left side do not work because they are based on matching happi containers. I think they should be based on matching happi device classes instead.
